### PR TITLE
chore: Improve typing in _polars/utils

### PR DESCRIPTION
**Pull request by MarcoGorelli**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/_polars/utils.py [hunk 1]
```diff
@@ -94,10 +95,10 @@ def native_to_narwhals_dtype(dtype: Any, dtypes: DTypes) -> DType:
...
 
 
-def narwhals_to_native_dtype(dtype: DType | type[DType], dtypes: DTypes) -> Any:
-    if (pl := get_polars()) is not None and isinstance(
-        dtype, (pl.DataType, pl.DataType.__class__)
-    ):
+def narwhals_to_native_dtype(dtype: DType | type[DType], dtypes: DTypes) -> pl.DataType:
+    import polars as pl  # ignore-banned-import()
+
+    if isinstance(dtype, (pl.DataType, pl.DataType.__class__)):  # type: ignore[arg-type]
```
Comment (MarcoGorelli) [inline narwhals/_polars/utils.py:101]: we're inside `_polars` here, so we shouldn't need `get_polars`, `polars` must already have been imported

File: narwhals/_polars/expr.py [hunk 2]
```diff
@@ -45,18 +45,18 @@ def func(*args: Any, **kwargs: Any) -> Any:
 
     def cast(self, dtype: DType) -> Self:
         expr = self._native_expr
-        dtype = narwhals_to_native_dtype(dtype, self._dtypes)
-        return self._from_native_expr(expr.cast(dtype))
+        dtype_pl = narwhals_to_native_dtype(dtype, self._dtypes)
+        return self._from_native_expr(expr.cast(dtype_pl))
```
Comment (MarcoGorelli) [inline narwhals/_polars/expr.py:49]: we need to rename these else mypy complains that we're assigning an incompatible type to an already-declared variable

File: narwhals/_polars/utils.py [hunk 3]
```diff
@@ -151,13 +150,13 @@ def narwhals_to_native_dtype(dtype: DType | type[DType], dtypes: DTypes) -> Any:
 
     if dtype == dtypes.List:  # pragma: no cover
         msg = "Converting to List dtype is not supported yet"
-        return NotImplementedError(msg)
+        raise NotImplementedError(msg)
```
[the remark this answers is not in the document]
Comment (FBruzzesi) [reply narwhals/_polars/utils.py:153]: How we missed that... in 3 places?!

[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply narwhals/_polars/utils.py:153]: 😄 it was probably me being too quick with this one - thanks for taking a look!

**Pull request merged**